akka/src/main/scala/cakesolutions/kafka/akka/KafkaConsumerActor.scala

```diff
@@ -10,22 +10,41 @@ import org.apache.kafka.common.serialization.Deserializer
 
 import scala.collection.JavaConversions._
 import scala.concurrent.duration._
-import scala.reflect.runtime.universe.{TypeTag, typeTag}
+import scala.reflect.runtime.universe.TypeTag
 
+/**
+  * An actor that wraps [[KafkaConsumerActor]].
+  *
+  * The actor pulls batches of messages from Kafka for all subscribed partitions,
+  * and forwards them to the supplied Akka actor reference in [[ConsumerRecords]] format.
+  *
+  * Before the actor continues pulling more data from Kafka,
+  * the receiver of the data must confirm the batches by sending back a [[KafkaConsumerActor.Confirm]] message
+  * that contains the offsets from the received batch.
+  * This mechanism allows the receiver to control the maximum rate of messages it will receive.
+  * By including the received offsets in the confirmation message,
+  * we avoid accidentally confirming batches that have not been fully processed yet.
```

Comment:
The Java Kafka consumer client has it turned on by default, which is why we have it turned on by default for `KafkaConsumer.Conf` as well. It definitely doesn't make sense for  `KafkaConsumerActor` to have it turned on by default. We probably want to always turn it off for `KafkaConsumerActor` regardless of the user setting.